src/test/java/co/rsk/bitcoinj/script/FlyoverRedeemScriptParserTest.java

```diff
@@ -0,0 +1,291 @@
+
+        erpRedeemScript = RedeemScriptUtils.createErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), erpRedeemScript);
+
+        p2shErpRedeemScript = RedeemScriptUtils.createP2shErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverP2shErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), p2shErpRedeemScript);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsStandardRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverStandardRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsErpRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverErpRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsP2shErpRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverP2shErpRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getM_whenFlyoverRedeemScriptContainsStandardRedeemScript_shouldReturnMValue() {
+        // Arrange
+        final int EXPECTED_M = 5;
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverStandardRedeemScript.getChunks());
+
+        // Act
+        int actualM = flyoverRedeemScriptParser.getM();
+
+        // Assert
+        assertEquals(EXPECTED_M, actualM);
+    }
+
+    @Test
+    public void getM_whenFlyoverRedeemScriptContainsErpRedeemScript_shouldReturnMValue() {
+        // Arrange
+        final int EXPECTED_M = 5;
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverErpRedeemScript.getChunks());
+
+        // Act
+        int actualM = flyoverRedeemScriptParser.getM();
+
+        // Assert
+        assertEquals(EXPECTED_M, actualM);
+    }
+
+    @Test
+    public void getM_whenFlyoverRedeemScriptContainsP2shErpRedeemScript_shouldReturnMValue() {
```

Comment:
Same, I think we should add some test cases testing against malformed scripts and non standard script

Reply:
Some methods such as the `getM` and `getMultiSigType` fail when the constructor, `FlyoverRedeemScriptParser`, has as an input a malformed script. Hence, it's not necessary to create specific methods with these test cases since they fail before executing these methods. The below test case tests this kind of case:

`flyoverRedeemScriptParser_whenRedeemScriptChunksIsMalformed_shouldThrowVerificationException`

On the other hand, a `NonStandardErpRedeemScriptParserHardcoded`, fails when you try to get the chunks with its `extractStandardRedeemScriptChunks` method, not when the constructor is executed, so it's not necessary to recreate this case.